test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -283,10 +297,54 @@ void ProtoConverter::visit(VarDecl const& _x)
 	m_output << "let " << varName << " := ";
 	visit(_x.expr());
 	m_output << "\n";
+	// If we are inside a for-init block, there are two places
+	// where the visited vardecl may have been defined:
+	// - directly inside the for-init block
+	// - inside a block within the for-init block
+	// In the latter case, we don't scope extend.
```

Comment:
This is controlled by assigning false to `m_forInitScopeExtEnabled` when we are already in a for-init block and  are creating a new block of statements or a statement (if/switch) that introduces a new block.